src/rust/engine/fs/src/lib.rs

```diff
@@ -459,9 +459,25 @@ pub struct PathGlobs {
   exclude: Arc<GitignoreStyleExcludes>,
   strict_match_behavior: StrictGlobMatching,
   conjunction: GlobExpansionConjunction,
+  patterns: Vec<glob::Pattern>,
```

Comment:
Given that very few PathGlobs are used for in-memory matching like this, it's probably not worth computing and storing this field for all of them.

IMO: would just swap back to the previous implementation (using the previous `matches` impl) and not worry about it. I was probably prematurely optimizing.

Reply:
The main reason to do this was to make `matches` return a `bool` rather than a `Result<bool, String>`, which makes it easier to use in the sorts of filtering contexts I'm using it in; it's not primarily a performance thing.